src/main.cc

```diff
@@ -13,30 +13,66 @@ int main(int argc, char** argv) {
     //! IO
     std::unique_ptr<IO> io(new IO(argc, argv));
 
-    //! Mesh
-    std::unique_ptr<Mesh<3, 8>> mesh(new GMSH<3, 8>());
-
     //! MaterialProperties
     std::shared_ptr<MaterialProperties> material =
```

Comment:
Always use `unique_ptr` unless it will have a shared ownership

Reply:
material properties are shared within material points. So has to be shared

Reply:
I find shared ownership of material properties difficult to understand. Feels not correct. Mesh owns material points, where the ownership is clear. Material points don't own material models. That's the problem I'm having. What's the best way? Maybe a const pointer to material model.